app/src/app/uploader/page.tsx

```diff
@@ -0,0 +1,148 @@
+'use client';
+import React, {useState} from 'react';
+import Papa from 'papaparse';
+import {ErrorNotification} from '../components/ErrorNotification';
+import {GerryDBViewSelector} from '../components/sidebar/GerryDBViewSelector';
+import {useMapStore} from '@/app/store/mapStore';
+import {Assignment, createMapDocument, patchUpdateAssignments} from '@/app/utils/api/apiHandlers';
+
+type MapLink = {
+  document_id: string;
+  name: string;
+};
+
+export default function Uploader() {
+  const [progress, setProgress] = useState<number>(0);
+  const [totalRows, setTotalRows] = useState<number>(0);
+  const [mapLinks, setMapLinks] = useState<MapLink[]>([]);
+
+  const ROWS_PER_BATCH = 2000;
+
+  const gTable = useMapStore(state => state.mapDocument?.gerrydb_table);
+
+  const handleDragOver = (event: React.DragEvent) => {
+    event.preventDefault();
+  };
+
+  const processFile = (file: File) => {
+    if (!file) {
+      throw new Error('No file selected');
+      return;
+    }
+
+    Papa.parse(file, {
+      complete: results => {
+        setProgress(0);
+        setTotalRows(results.data.length);
+
+        createMapDocument({
+          gerrydb_table: gTable ?? '',
+        }).then(response => {
+          const {document_id} = response;
+          let rowCursor = 0;
+
+          const partialUploadStep = () => {
+            const assignments: Assignment[] = [];
+            const rows = results.data as Array<Array<string>>;
+            rows.slice(rowCursor, rowCursor + ROWS_PER_BATCH).forEach(row => {
+              if (row.length == 2 && row[1] !== '' && !isNaN(Number(row[1]))) {
+                assignments.push({document_id, geo_id: row[0], zone: Number(row[1])});
+              }
+            });
+            patchUpdateAssignments(assignments).then(stepResult => {
+              setProgress(rowCursor + assignments.length);
+              rowCursor += ROWS_PER_BATCH;
+              if (rowCursor > results.data.length) {
+                setMapLinks([...mapLinks, {document_id, name: file.name}]);
+              } else {
+                setTimeout(partialUploadStep, 10);
+              }
+            });
+          };
+          setProgress(0);
+          partialUploadStep();
+        });
+      },
+    });
+  };
+
+  const handleDrop = (event: React.DragEvent) => {
+    event.preventDefault();
+    const file = event.dataTransfer.files?.[0];
+    processFile(file);
+  };
+
+  const handleFileSelected = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      processFile(file);
+    }
+  };
+
+  return (
+    <div className="h-screen w-screen flex items-center justify-center bg-gray-100">
```

Comment:
Medium (?): Use Radix UI Themes component (`<Flex>`, `<Heading>`, `<Box>` etc) when possible to ensure styling consistency